source/SIL.AppBuilder.Portal.Frontend/src/ui/components/header/notifications/index.tsx

```diff
@@ -125,52 +56,44 @@ class Notifications extends React.Component<IProps> {
         <div
           data-test-notification-trigger
           style={{ position: 'relative' }}
-          onClick={this.toggle}>
+          onClick={this.toggle}
+        >
           {!seenNotifications && <div className='red-dot' />}
           <i className='alarm large circular icon no-shadows' />
         </div>
 
         <div className={`ui menu transition notifications ${this.state.visible ? 'visible' : ''}`}>
           {
-            notifications && notifications.length > 0 && this.showNotifications() ?
+            notifications && notifications.length > 0 && showNotifications ?
             <>
               <div className="notification-buttons">
                 <a href="#" onClick={e => {
                   e.preventDefault();
-                  this.clearAll(e);
+                   clearAll();
                   }}
                 >
                   {t('header.clearAll') }
                 </a>
               </div>
               <div className={notifications.length > 3 ? 'scrollable-menu' : ''}>
                 {
-                  notifications.sort((a, b) => {
-
-                    return (a.attributes.time === b.attributes.time) ? 0 :
-                      a.attributes.time > b.attributes.time ? -1 : 1;
-
-                  }).sort((a, b) => {
-
-                    return (a.attributes.isViewed === b.attributes.isViewed) ? 0 :
-                      a.attributes.isViewed ? 1 : -1;
-
-                  }).map((notification, index) => {
+                  notifications.map((notification, index) => {
```

Comment:
There should maybe be a separate component for each notification. That'll clean this up a bit :)

Reply:
right on. :+1: